tests/test_shadow_run_id.py

```
@@ -0,0 +1,109 @@
+class TestRunShadowUsesIt:
+    def test_orders_are_tagged_with_a_shadow_id(self, tmp_path, monkeypatch):
+        ids = _one_session(tmp_path, monkeypatch, "a.db")
+        assert ids, "the session wrote no orders, so it proves nothing"
+        assert all(i.startswith("shadow-") for i in ids), ids
+
+    def test_two_sessions_do_not_share_an_id(self, tmp_path, monkeypatch):
+        # The exact failure: two 30-minute rehearsals 3.85 hours apart both
+        # wrote as run-2809a7161de1 and could only be separated by eye.
+        first = _one_session(tmp_path, monkeypatch, "first.db")
+        second = _one_session(tmp_path, monkeypatch, "second.db")
+        assert first and second
```

Ruff (0.16.2)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)